src/pathedit.cpp

```diff
@@ -118,7 +120,23 @@ bool PathEdit::event(QEvent* e) {
     return QLineEdit::event(e);
 }
 
+void PathEdit::onTextEdited(const QString& text) {
+    // just replace start tilde with home path if text is changed by user
+    if(text == QLatin1String("~") || text.startsWith(QLatin1String("~/"))) {
+        QString txt(text);
+        txt.replace(0, 1, QDir::homePath());
```

Comment:
Just a note... there is a API in libfm-qt for the home path -> `Fm::FilePath::homeDir()`